ctapipe/tools/stage1.py

```diff
@@ -0,0 +1,750 @@
+                    )
+
+    def _write_instrument_configuration(self, subarray):
+        """write the SubarrayDescription
+
+        Parameters
+        ----------
+        subarray : ctapipe.instrument.SubarrayDescription
+            subarray description
+        """
+        self.log.debug("Writing instrument configuration")
+        serialize_meta = True
+
+        subarray.to_table().write(
+            self.output_path,
+            path="/configuration/instrument/subarray/layout",
+            serialize_meta=serialize_meta,
+            append=True,
+        )
+        subarray.to_table(kind="optics").write(
+            self.output_path,
+            path="/configuration/instrument/telescope/optics",
+            append=True,
+            serialize_meta=serialize_meta,
+        )
+        for telescope_type in subarray.telescope_types:
+            ids = set(subarray.get_tel_ids_for_type(telescope_type))
+            if len(ids) > 0:  # only write if there is a telescope with this camera
+                tel_id = list(ids)[0]
+                camera = subarray.tel[tel_id].camera
+                camera.geometry.to_table().write(
+                    self.output_path,
+                    path=f"/configuration/instrument/telescope/camera/geometry_{camera}",
+                    append=True,
+                    serialize_meta=serialize_meta,
+                )
+                camera.readout.to_table().write(
+                    self.output_path,
+                    path=f"/configuration/instrument/telescope/camera/readout_{camera}",
+                    append=True,
+                    serialize_meta=serialize_meta,
+                )
+
+    def _write_processing_statistics(self):
+        """ write out the event selection stats, etc. """
+        image_stats = self.check_image.to_table(functions=True)
+        image_stats.write(
+            self.output_path,
+            path="/dl1/service/image_statistics",
+            append=True,
+            serialize_meta=True,
+        )
+
+    def _parameterize_image(self, tel_id, image, signal_pixels, peak_time=None):
+        """Apply image cleaning and calculate image features
+
+        Parameters
+        ----------
+        subarray : SubarrayDescription
+           subarray description
+        data : DL1CameraContainer
+            calibrated camera data
+        tel_id: int
+            which telescope is being cleaned
+
+        Returns
+        -------
+        np.ndarray, ImageParametersContainer:
+            cleaning mask, parameters
+        """
+
+        tel = self.event_source.subarray.tel[tel_id]
+        geometry = tel.camera.geometry
+        image_selected = image[signal_pixels]
+
+        # check if image can be parameterized:
+        image_criteria = self.check_image(image_selected)
+        self.log.debug(
+            "image_criteria: %s",
+            list(zip(self.check_image.criteria_names[1:], image_criteria)),
```

Comment:
would it be interesting to write the criteria status event per event as columns in the DL1 parameters table? overkill?